morpheus/_lib/doca/src/doca_source_kernels.cu

```diff
@@ -157,247 +190,271 @@ __device__ int64_t mac_bytes_to_int64(uint8_t* mac)
+                _payload_sizes[i] = payload_size;
+                _payload_flags[i] = 1;
+                pkt_info->payload_sizes_out[packet_idx] = payload_size;
+                // mac address
+                pkt_info->src_mac_out[packet_idx] = mac_bytes_to_int64(hdr_udp->l2_hdr.s_addr_bytes);
+                pkt_info->dst_mac_out[packet_idx] = mac_bytes_to_int64(hdr_udp->l2_hdr.d_addr_bytes);
+                // ip address
+                pkt_info->src_ip_out[packet_idx] = ip_to_int64(hdr_udp->l3_hdr.src_addr);
+                pkt_info->dst_ip_out[packet_idx] = ip_to_int64(hdr_udp->l3_hdr.dst_addr);
+                // ports
+                pkt_info->src_port_out[packet_idx] = BYTE_SWAP16(hdr_udp->l4_hdr.src_port);
+                pkt_info->dst_port_out[packet_idx] = BYTE_SWAP16(hdr_udp->l4_hdr.dst_port);
+                // frame type
+                pkt_info->ether_type_out[packet_idx] = static_cast<int32_t> (hdr_udp->l2_hdr.ether_type);
+                // protocol id
+                pkt_info->next_proto_id_out[packet_idx] = static_cast<int32_t> (hdr_udp->l3_hdr.next_proto_id);
+            }
+
+            auto now = cuda::std::chrono::system_clock::now();
+            auto now_ms = cuda::std::chrono::time_point_cast<cuda::std::chrono::milliseconds>(now);
+            auto epoch = now_ms.time_since_epoch();
+            pkt_info->timestamp_out[packet_idx] = epoch.count();
+        }
+        
+        // if (threadIdx.x == 0) DEVICE_GET_TIME(reduce_start);
+        auto payload_size_total = BlockReduce(temp_storage).Sum(_payload_sizes);
+        __syncthreads();
+        auto packet_count = BlockReduce(temp_storage).Sum(_payload_flags);
+        __syncthreads();
+        // if (threadIdx.x == 0) DEVICE_GET_TIME(reduce_stop);
+
+        if (threadIdx.x == 0) {
+            // DEVICE_GET_TIME(pkt_proc);
+            DOCA_GPUNETIO_VOLATILE(pkt_info->packet_count_out) = packet_count;
+            DOCA_GPUNETIO_VOLATILE(pkt_info->payload_size_total_out) = payload_size_total;
+            // printf("Block %d Update semaphore %d with %d packets %d size\n",
+            //   blockIdx.x,
+            //   sem_idx, DOCA_GPUNETIO_VOLATILE(pkt_info->packet_count_out), DOCA_GPUNETIO_VOLATILE(pkt_info->payload_size_total_out));
+            doca_ret = doca_gpu_dev_semaphore_set_status(sem, sem_idx, DOCA_GPU_SEMAPHORE_STATUS_READY);
+            if (doca_ret != DOCA_SUCCESS) {
+                printf("Error %d doca_gpu_dev_semaphore_set_status\n", doca_ret);
+                DOCA_GPUNETIO_VOLATILE(*exit_condition) = 1;
+                // break;
+            }
+
+            // printf("CUDA rx time %ld proc time %ld pkt conv %ld block reduce %ld\n",
+            //         rx_stop - rx_start,
+            //         pkt_proc - rx_stop,
+            //         reduce_start - rx_stop,
+            //         reduce_stop - reduce_start);
+        }
+        __syncthreads();
+
+#if 0
```

Comment:
Please delete unused code.

Reply:
added an if macro as this is the code to run it persistently (which can be useful to turn on for debug)